compel/src/lib/ptrace.c

```diff
@@ -110,18 +110,26 @@ int ptrace_swap_area(pid_t pid, void *dst, void *src, long bytes)
 }
 
 int __attribute__((weak)) ptrace_get_regs(int pid, user_regs_struct_t *regs) {
+#ifdef CONFIG_MIPS
+     	int ret = ptrace(PTRACE_GETREGS, pid, NULL, regs);
+#else
```

Comment:
If `ptrace(PTRACE_GETREGSET)` does not work for MIPS for some reason, you would need to override the entire `ptrace_{get,set}_regs` in `compel/arch/mips/src/lib/infect.c`. See for example as s390 and x86 do it.